internal/cmd/cmd.go

```diff
@@ -270,27 +283,40 @@ func (c *Cmd) Wait() error {
 		state.exited = true
 		state.code = code
 	}
+
 	// Terminate the IO if the copy does not complete in the requested time.
+	timeoutErrCh := make(chan error)
```

Comment:
Instead of having a separate channel for timeout error, can we just have a separate error variable that will be set only in the `<-t.C:` block?  That way we don't need to worry about closing the channel and the following code block:

```
if tErr := <-timeoutErrCh; ioErr == nil {
    ioErr = tErr
}
```

can just be 

```
if ioErr == nil {
    ioErr = tErr
}
```

This approach seems easier to read/understand for me at least. What do you think?

Reply:
we need to block on `timeoutErrCh` regardless, so we need some mechanism to have `Wait` actually wait for the timeout

the alternatives (imo) would be to add a function that `selects` on `<-allDoneCh` and `ctx.Done()`, and pass it a context with the `CopyAfterExitTimeout` or something similar, which i would rather save for when we refactor `Cmd` wholesale to properly thread contexts everywhere